src/pipeline/node/ToF.cpp

```
@@ -1,10 +1,135 @@
+}
+
+bool isSocketAlreadyConnected(const Pipeline& pipeline, CameraBoardSocket boardSocket) {
+    for(const auto& node : pipeline.getAllNodes()) {
+        auto camera = std::dynamic_pointer_cast<Camera>(node);
+        if(!camera) {
+            continue;
+        }
+        try {
+            if(camera->getBoardSocket() == boardSocket) {
+                return true;
+            }
```

Cppcheck (2.21.0)

[error] There is an unknown macro here somewhere. Configuration is required. If DEPTHAI_NLOHMANN_DEFINE_TYPE_INTRUSIVE is a macro then please configure it.

(unknownMacro)